c_tests/tests/compile_trace_not_entry_bb.c

```diff
@@ -0,0 +1,34 @@
+// Compiler:
+// Run-time:
+
+// Check that trace compilation works in the non-entry block.
+//
+// Why? Because LLVM allocas typically appear in the entry block, so tracing in
+// another block will means they are not seen in the trace.
```

Comment:
s/means/mean.

But maybe change this description to something like:

Since LLVM allocas only seem to appear in the entry block of a function, we will miss the allocas if tracing starts in a later block.